db/size_estimates_virtual_reader.cc

```diff
@@ -208,10 +208,10 @@ static future<std::vector<token_range>> get_local_ranges(database& db) {
         // All queries will be on that table, where all entries are text and there's no notion of
         // token ranges form the CQL point of view.
         auto left_inf = boost::find_if(ranges, [] (auto&& r) {
-            return r.end() && (!r.start() || r.start()->value() == dht::minimum_token());
+            return r.end() && (!r.start() || r.start()->value().is_minimum());
         });
         auto right_inf = boost::find_if(ranges, [] (auto&& r) {
-            return r.start() && (!r.end() || r.end()->value() == dht::maximum_token());
+            return r.start() && (!r.end() || r.end()->value().is_maximum());
```

Comment:
This can lead to an unintended change: if one of the tokens happened to be the minimum or maximum, we'd misdetect it as left_inf or right_inf.

However, I don't think get_primary_ranges_for() generates ranges with minimum_token - it leads to interval::unwrap() which uses half-open intervals instead. So I think this can just be dropped.

Reply:
I don't understand. This particular diff doesn't change behaviour. `== dht::maximum_token()` does the same thing as `is_maximum()`. The goal of this series so far is to remove `dht::maximum_token` first (which will ensure that we aren't producing +inf tokens, but we may still be receiving them through IDL), and then remove +inf tokens altogether (and then this diff will be revised).

Reply:
Maybe I was taking into account a step that hasn't happened yet, switching maximum_token to greatest_token(). If you don't take that step, and the intent is just to retire maximum_token(), then there is no risk.